docs/customizing-makefiles.rst

```diff
@@ -2,236 +2,95 @@ Customizing the Makefiles
-   :emphasize-lines: 3, 4, 5, 6, 9, 10, 22, 25, 28, 31
-   :linenos:
-
-   mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
-   current_dir := $(patsubst %/,%,$(dir $(mkfile_path))) 
-   TOP := top
-   VERILOG := ${current_dir}/counter.v 
-   DEVICE  := xc7a50t_test
-   BITSTREAM_DEVICE := artix7
-   BUILDDIR := build
-
-   PARTNAME := xc7a35tcpg236-1
-   XDC := ${current_dir}/basys3.xdc 
-   BOARD_BUILDDIR := ${BUILDDIR}/basys3
-
-
-   .DELETE_ON_ERROR:
-
-   all: ${BOARD_BUILDDIR}/${TOP}.bit
-
-   ${BOARD_BUILDDIR}:
-      mkdir -p ${BOARD_BUILDDIR}
-
-   ${BOARD_BUILDDIR}/${TOP}.eblif: | ${BOARD_BUILDDIR}
-      cd ${BOARD_BUILDDIR} && symbiflow_synth -t ${TOP} -v ${VERILOG} -d ${BITSTREAM_DEVICE} -p ${PARTNAME} -x ${XDC} 2>&1 > /dev/null
-
-   ${BOARD_BUILDDIR}/${TOP}.net: ${BOARD_BUILDDIR}/${TOP}.eblif
-      cd ${BOARD_BUILDDIR} && symbiflow_pack -e ${TOP}.eblif -d ${DEVICE} 2>&1 > /dev/null
-
-   ${BOARD_BUILDDIR}/${TOP}.place: ${BOARD_BUILDDIR}/${TOP}.net
-      cd ${BOARD_BUILDDIR} && symbiflow_place -e ${TOP}.eblif -d ${DEVICE} -n ${TOP}.net -P ${PARTNAME} 2>&1 > /dev/null
+To download Symbiflows Makefile.common, run the following command inside the directory containing your 
+build files:
 
-   ${BOARD_BUILDDIR}/${TOP}.route: ${BOARD_BUILDDIR}/${TOP}.place
-      cd ${BOARD_BUILDDIR} && symbiflow_route -e ${TOP}.eblif -d ${DEVICE} 2>&1 > /dev/null
-
-   ${BOARD_BUILDDIR}/${TOP}.fasm: ${BOARD_BUILDDIR}/${TOP}.route
-      cd ${BOARD_BUILDDIR} && symbiflow_write_fasm -e ${TOP}.eblif -d ${DEVICE}
-
-   ${BOARD_BUILDDIR}/${TOP}.bit: ${BOARD_BUILDDIR}/${TOP}.fasm
-      cd ${BOARD_BUILDDIR} && symbiflow_write_bitstream -d ${BITSTREAM_DEVICE} -f ${TOP}.fasm -p ${PARTNAME} -b ${TOP}.bit
+.. code-block:: bash
 
-   clean:
-      rm -rf ${BUILDDIR}
+   wget https://raw.githubusercontent.com/SymbiFlow/symbiflow-examples/master/common/Makefile.common
 
+Then create a main makefile by running ``touch Makefile``, and add the following to the contents.
+Don't change the two highlighted lines unless you know what you are doing:
 
-Adding HDL files to your design 
---------------------------------
+.. code-block:: bash
+   :name: makefile-example
+   :emphasize-lines: 1, 9
+   :linenos:
 
-:ref:`Line 3 <makefile-example>` in the Makefile shows how to define the name for your top level module. For example, if
-your top module was named ``module switches ( ...``  then you would simply change line 3 to 
-``TOP := switches``.
+   current_dir := ${CURDIR}
+   TOP := <put the name of your top module here>
+   SOURCES := ${current_dir}/<put your HDL sources here>
+   
+   # Include your constraint file path(s) below. Use either an XDC file 
+   # or a PCF+SDC pair. Don't use all three file types.
+   XDC := ${current_dir}/<name of your pcf file if applicable>
+   PCF := ${current_dir}/<name of your xdc file if applicable>
+   SDC := ${current_dir}/<name of your sdc file if applicable>
 
-.. warning:: 
+   include ${current_dir}/Makefile.common 
```

Comment:
I believe this is not the correct location of the common makefile. what about sth like:

```
include <symbiflow-examples root dir>/common/Makefile.common
```

Reply:
`common.mk` is a pretty typical name...